### PR TITLE
DEV: Fix pending posts page, make tests work in legacy env

### PR DESCRIPTION
**Pull request by CvX**
* Running the tests only in the ember cli env hid the fact that the pending posts feature wasn't working in the legacy environment
* Tests were using ember-cli-only APIs while there are widely used testing APIs in Discourse that support both ember envs
* `ember-test-selectors` was in both dependencies and devDependencies in discourse/package.json
* `qunit-dom` in package.json was not only unused but also defunct, as it wasn't pulled into the legacy env app

A followup to #14501, and #15128.

### CODE REVIEW

**Comment by Flink** [comment]
It was my understanding that we want to drop the legacy env anyway so we should probably focus more on doing things the ember-cli way™ :thinking: It’s been discussed briefly here regarding tests: https://github.com/discourse/discourse/pull/14501#discussion_r752322427

**Comment by Flink** [comment]
Just understood that it seems Ember CLI is not enabled in production and that’s what causing this bug: https://meta.discourse.org/t/can-i-view-my-own-topic-that-is-awaiting-moderation/187532/9
I wasn’t aware of that, so sad :sweat: 

So thanks for the fix Jarek! :slightly_smiling_face:

File: app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js [hunk 1]
```diff
@@ -1,28 +1,20 @@
...
+    const container = document.getElementById("ember-testing-container");
+    container.scrollTo(0, 0);
+
     await visit("/t/internationalization-localization/280");
+    container.scrollTo(0, 800);
+    container.scrollTo(0, 700);
 
-    await scrollTo(container, 0, 800);
-    await scrollTo(container, 0, 700);
+    // await waitUntil(() => find(".sticky-avatar"));
```
Comment (Flink) [inline app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js:14]: Leftover?

File: app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js [hunk 2]
```diff
@@ -1,28 +1,20 @@
...
-
-  const container = document.getElementById("ember-testing-container");
-
-  needs.hooks.beforeEach(function () {
-    container.scrollTop = 0;
-  });
+import { visit } from "@ember/test-helpers";
 
+acceptance("Sticky Avatars", function () {
   skip("Adds sticky avatars when scrolling up", async function (assert) {
```
Comment (Flink) [inline app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js:6]: This test is actually skipped :sweat_smile:

**Comment by CvX** [comment]
For now just a handful of production sites run on ember-cli build and it's still considered kind of experimental, so we have to support both envs. 🙂

File: app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js (hunk 2 above)
Comment (CvX) [reply app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js:6]: Yup! I tried to fix it and un-skip it, but it didn't work out 🙂 At least it's in a better shape for the next person that want to take a stab at it. (though I suppose it's more likely this test will be deleted 😛)

**Review by Flink — APPROVED** [review]

**Pull request merged**